library/core/src/slice/index.rs

```diff
@@ -214,7 +214,8 @@ unsafe impl<T> SliceIndex<[T]> for usize {
     #[inline]
     fn get(self, slice: &[T]) -> Option<&T> {
         // SAFETY: `self` is checked to be in bounds.
-        if self < slice.len() { unsafe { Some(&*self.get_unchecked(slice)) } } else { None }
+        // The call to `add` is safe, as we check the bounds.
+        if self < slice.len() { unsafe { Some(&*slice.as_ptr().add(self)) } } else { None }
```

Comment:
this could be a perf win on its own, so testing this together with the other change isn't super useful.

Reply:
Do you want to try that change on its own?